docs/data/material/guides/routing/LinkRouterWithTheme.tsx

```diff
@@ -33,7 +33,7 @@ const theme = createTheme({
     MuiLink: {
       defaultProps: {
         component: LinkBehavior,
-      } as LinkProps,
+      } as LinkProps<typeof LinkBehavior>,
```

Comment:
I think we don't need to cast now as `component` type is available.
```suggestion
      },
```

Reply:
fixed here https://github.com/mui/material-ui/pull/35924/commits/84ecfff03d5832b911055582004eabb8714a39b3